src/json/serialization.cr

```diff
@@ -270,60 +270,70 @@ module JSON
         {% properties = {} of Nil => Nil %}
         {% for ivar in @type.instance_vars %}
           {% ann = ivar.annotation(::JSON::Field) %}
-          {% unless ann && (ann[:ignore] || ann[:ignore_serialize]) %}
+          {% unless ann && ann[:ignore] %}
             {%
               properties[ivar.id] = {
-                type:      ivar.type,
-                key:       ((ann && ann[:key]) || ivar).id.stringify,
-                root:      ann && ann[:root],
-                converter: ann && ann[:converter],
-                emit_null: (ann && (ann[:emit_null] != nil) ? ann[:emit_null] : emit_nulls),
+                type:             ivar.type,
+                key:              ((ann && ann[:key]) || ivar).id.stringify,
+                root:             ann && ann[:root],
+                converter:        ann && ann[:converter],
+                emit_null:        (ann && (ann[:emit_null] != nil) ? ann[:emit_null] : emit_nulls),
+                ignore_serialize: ann && ann[:ignore_serialize],
               }
             %}
           {% end %}
         {% end %}
 
         json.object do
           {% for name, value in properties %}
-            _{{name}} = @{{name}}
+            {% unless value[:ignore_serialize] == true %}
+              _{{name}} = @{{name}}
 
-            {% unless value[:emit_null] %}
-              unless _{{name}}.nil?
-            {% end %}
-
-              json.field({{value[:key]}}) do
-                {% if value[:root] %}
-                  {% if value[:emit_null] %}
-                    if _{{name}}.nil?
-                      nil.to_json(json)
-                    else
-                  {% end %}
+              {% unless value[:ignore_serialize].nil? || value[:ignore_serialize] == false %}
```

Comment:
I believe this could just be `if value[:ignore_serialize]`. Same semantics but more concise.
```suggestion
              {% if value[:ignore_serialize] %}
```